chainreg/chainregistry.go

```diff
@@ -190,44 +164,30 @@ var DefaultLtcChannelConstraints = channeldb.ChannelConstraints{
 	MaxAcceptedHtlcs: input.MaxHTLCNumber / 2,
 }
 
-// ChainControl couples the three primary interfaces lnd utilizes for a
-// particular chain together. A single ChainControl instance will exist for all
-// the chains lnd is currently active on.
-type ChainControl struct {
-	// ChainIO represents an abstraction over a source that can query the blockchain.
-	ChainIO lnwallet.BlockChainIO
+// PartialChainControl contains all the primary interfaces of the chain control
+// that can be purely constructed from the global configuration. No wallet
+// instance is required for constructing this partial state.
+type PartialChainControl struct {
```

Comment:
(useless) nit: Not sure about `PartialChainControl` as a name, but also can't think of a better one (`Common`/`Global`..?)

Reply:
Yeah, naming is hard... But I think `partial` isn't too bad of a name here since we split the initialization into two parts, so it's only complete after the second call.